src/systems/filecoin_files/data_transfer/_index.md

```diff
@@ -3,4 +3,92 @@ menuTitle: Data Transfer
+4. The responder's data transfer module schedules the transfer
+5. The responder makes a GraphSync request for the data
+6. The requestor receives the graphsync request, verifies it's in the scheduler and begins sending data
+7. The responder receives data and can produce an indication of progress
+8. The responder completes receiving data, and notifies any listeners
+
+The push flow is ideal for storage deals, where the client initiates the push
+once it verifies the the deal is signed and on chain
+
+## Pull Flow
+
+{{< diagram src="./docs/systems/filecoin_files/data_transfer/pull-flow.mmd.svg" title="Data Transfer - Pull Flow" >}}
+
+1. A requestor initiates a Pull transfer when it wants to receive data from another party. 
+2. The requestors' data transfer module will send a pull request to the responder along with the data transfer voucher. 
+3. The responder's data transfer module validates the data transfer request via a PullValidator provided as a dependency by the responder
+4. The responder's data transfer module schedules the transfer (meaning it is expecting the requestor to initiate the actual transfer)
+5. The responder's data transfer module sends a response to the requestor saying it has accepted the transfer and is waiting for the requestor to initiate the transfer
+6. The requestor schedules the data transfer
+7. The requestor makes a GraphSync request for the data
+8. The responder receives the graphsync request, verifies it's in the scheduler and begins sending data
+9. The requestor receives data and can produce an indication of progress
+10. The requestor completes receiving data, and notifies any listeners
+
+The pull flow is ideal for retrieval deals, where the client initiates the pull when the deal is agreed upon.
+
+# Alternater Pull Flow - Single Round Trip
+
+{{< diagram src="./docs/systems/filecoin_files/data_transfer/alternate-pull-flow.mmd.svg" title="Data Transfer - Single Round Trip Pull Flow" >}}
+
+1. A requestor initiates a Pull transfer when it wants to receive data from another party. 
+2. The requestor’s DTM schedules the data transfer
+3. The requestor makes a Graphsync request to the responder with a data transfer request
+4. The responder receives the graphsync request, and forwards the data transfer request to the data transfer module
+5. The requestors' data transfer module will send a pull request to the responder along with the data transfer voucher. 
+6. The responder's data transfer module validates the data transfer request via a PullValidator provided as a dependency by the responder
+7. The responder's data transfer module schedules the transfer
+8. The responder sends a graphsync response along with a data transfer accepted response piggypacked
+9. The requestor receives data and can produce an indication of progress
+10. The requestor completes receiving data, and notifies any listeners
```

Comment:
this 1-RTT flow is great, LGTM 👍

Reply:
- this reminded me of a discussion at filecoin team week, that maybe the validators should be plugged into graphsync instead of DT? or that maybe DT is a "more advanced graphsync module w/ scheduler and validation awareness" (kind of like a "more advanced HTTP Client or HTTP Server with connection pools and pipelining or whatever else"). 
- curious if this connection seems better or worse to you + how you think about graphsync?  it may be worthwhile to just make this module be "part of graphsync"
- (eg when i think about go-ipfs using graphsync, i think of it trying to plug in some validators (eg accept pushes if i've friended this node, or part of the same ipfs-cluster, or if bitswap ratio is X, or ...)